### ISSUE TITLE
Windows Support Virtually Nonexistent

### ISSUE THREAD

**Issue by yash101** [yash101/ApplicationServer#4]
Windows support = 0.0% ATM.

**Comment by zachw000** [comment]
Would it be possible to run this off of Cygwin? What POSIX and UNIX/Linux specific functions are you using and implementing in the development of this server?

**Comment by yash101** [comment]
Yes! Cygwin would work perfectly. However, I want to build native support,
automatically toggled using the C++ preprocessor.
To do this, I will be adding Windows POSIX subsystem support for sockets
and I will be disabling sendfile() if I can't find an alternative.
sendfile() is very fast and powerful, but I don't think it exists in all
platforms.
It should compile perfectly under Cygwin, currently. Be my guest to try it,
as I have not tested it. I do not use system calls such as fork() because
they are very heavy on non-GNU Linux platforms.
I use C++11 features wherever possible so most code is standardized.

Thank You,
_Devyash Lodha_
http://devyashis.me/

**Comment by yash101** [comment]
Functions being used (roughly):
- send();
- recv();
- sendfile();
- socket();
- bind();
- listen();
- setsocketopt();
- bind();

Libraries being used (roughly):
- pthread (STATIC) (`-pthread`)
- Automatically-injected libs by G++/Clang++

**Comment by yash101** [comment]
Windows support is now 50%.
Code compiles completely; a few warnings are issued, but cannot be mitigated due to the design of this server which give it high efficiency.

The application launches in Windows, but crashes immediately after launch during the static initialization of the statically-declared objects.

Error breakdown is in issue #7 .

**Comment by zachw000** [comment]
Can you run a debugger on it, and see where it is breaking? After that I could try working that out for you.
